File: back/DevC/src/main/java/com/back/devc/domain/post/category/entity/Category.java
package com.back.devc.domain.post.category.entity;

import com.back.devc.domain.post.post.entity.Post;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long categoryId;

    @Column(nullable = false, unique = true, length = 50)
    private String name;

    //one to many 처리 완료
    @OneToMany(mappedBy = "category",fetch = FetchType.LAZY)
    private List<Post> posts = new ArrayList<>();

    // 생성자
    public Category(String name) {
        this.name = name;
    }

    // 이름 수정 (필요하면)
    public void update(String name) {
        this.name = name;
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/category/entity/Category.kt
package com.back.devc.domain.post.category.entity

import com.back.devc.domain.post.post.entity.Post
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity
class Category(

    @Column(nullable = false, unique = true, length = 50)
    private var name: String
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val categoryId: Long? = null

    @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
    private val posts: MutableList<Post> = mutableListOf()

    // JPA 기본 생성자용 (필수)
    protected constructor() : this(name = "")

    fun update(name: String) {
        this.name = name
    }

    fun getName(): String = name

    fun getPosts(): List<Post> = posts.toList()
}